acstools/acszpt.py

```diff
@@ -268,116 +267,86 @@ def _check_date(self, fmt='%Y-%m-%d'):
+                invokeURL = "https://vtbopx9sf3.execute-api.us-east-1.amazonaws.com/main/single-filter-resource"  # noqa
+                # Generate a request body to send the API
+                body = '{"date": "%s", "detector": "%s", "filter": "%s"}' % (self.date, self.detector, self.filt)
 
-        # Grab warnings returned by the ZPT calc as marked by stylized h4 tag
-        warnings = soup.find_all('h4', {'style': "color:red;"})
+            # send request to APIGateway with try/except clauses
+            help_desk_string = ("\nIf this error persists, please contact the HST Help Desk at \n"
+                                "https://stsci.service-now.com/hst")
 
-        # Remove tags and 'Warnings: ' label attached to warnings
-        str_warnings = [w.get_text().replace('Warnings: ', '') for w in warnings]
+            try:
+                response = requests.post(invokeURL, data=body)
 
-        # Add non-empty strings to _warnings attribute
-        self._warnings = [w for w in str_warnings if w.strip()]
+            except requests.exceptions.HTTPError:
+                err_string = "HTTP Error to AWS API Gateway:" + help_desk_string
+                print(err_string)
 
-        # Turn the single list into a 2-D numpy array
-        data = np.reshape(td,
-                          (int(len(td) / self._block_size), self._block_size))
-        # Create the QTable, note that sometimes self._response will be empty
-        # even though the return was successful; hence the try/except to catch
-        # any potential index errors. Provide the user with a message and
-        # set the zpt_table to None.
-        try:
-            tab = QTable(data[1:, :],
-                         names=data[0],
-                         dtype=[str, float, float, float, float, float])
-        except IndexError as e:
-            msg = (f'{repr(e)}\n{self._msg_div}\n'
-                   'There was an issue parsing the request. '
-                   'Try resubmitting the query. If this issue persists, please '
-                   'submit a ticket to the Help Desk at'
-                   'https://stsci.service-now.com/hst')
-            LOG.info(msg)
-            self._zpt_table = None
-        else:
-            # If and only if no exception was raised, attach the units to each
-            # column of the QTable. Note we skip the FILTER column because
-            # Quantity objects in astropy must be numerical (i.e. not str)
-            for col in tab.colnames:
-                if col.lower() == 'filter':
-                    continue
-                tab[col].unit = self._data_units[col]
+            except requests.exceptions.ConnectionError:
+                err_string = "Error Connecting to AWS API Gateway:" + help_desk_string
+                print(err_string)
 
-            self._zpt_table = tab
+            except requests.exceptions.Timeout:
+                print("Timeout Error to AWS API Gateway:" + help_desk_string)
```

Comment:
Not sure if you are done editing or not but you should not be printing, you should re-raise exception.

Reply:
not done yet sorry, but yes I will adjust that